Include new logo into the frontpage and adapt according #174

zbarovsky opened the pull request:
Fixes #174 
Fixes #175 

Here are the following changes I made:

- Added the new logo where the old one was (index.html line 55)
- updated the following svgs to match recommended color palette: globe_logo.svg, and logo_world_driven.svg
- I also went into the style.css file and updated input:focus + .slider (lines 126-132) and changed the color to also match the color palette provided.

If I didn't do what was needed, or messed with something I wasn't supposed to (i.e. the css file), please let me know and I can fix it.

Also, for some reason I was having trouble looking at the index.html imgs when the full src tag was there (i.e. /static/images/...) and so when working I removed those to see, but added them back after the fact. If they don't load for any reason, let me know and I can fix it too.

TooAngel commented on static/images/globe_logo.svg:
```diff
@@ -1,7 +1,7 @@
 <circle cx="42.5" cy="42.5" r="42.5" fill="#C9D4D9"/>
 <mask id="mask0" mask-type="alpha" maskUnits="userSpaceOnUse" x="0" y="0" width="85" height="85">
-<circle cx="42.5" cy="42.5" r="42.5" fill="#2591C2"/>
+<circle cx="42.5" cy="42.5" r="42.5" fill="#205572"/>
```
In my opinion, there is no need to change the color in the log itself.

In the issue, I thought more about applying the color schema if possible to the rest of the page.

TooAngel commented on static/images/logo_world_driven.svg:
```diff
@@ -25,6 +25,6 @@
 <path d="M60.5 100C39 86.5 34 102.5 16 109.501C23.6667 118.335 40.8 136.502 48 138.502C55.2 140.502 90.3333 151.667 107 156.999C121.833 131.166 124.9 74.8016 112.5 88.0016C97 104.502 82 113.5 60.5 100Z" fill="#205572"/>
 </g>
-<path d="M165.684 35.872L166.892 36.1934C166.992 35.8183 166.912 35.418 166.675 35.1102C166.438 34.8024 166.072 34.622 165.684 34.622V35.872ZM155.804 73V74.25C156.371 74.25 156.866 73.8689 157.012 73.3214L155.804 73ZM151.072 73L149.874 73.3559C150.031 73.8863 150.519 74.25 151.072 74.25V73ZM143.844 48.664L145.042 48.3081C145.04 48.2992 145.037 48.2903 145.034 48.2815L143.844 48.664ZM142.96 45.7L1 ...
+<path d="M165.684 35.872L166.892 36.1934C166.992 35.8183 166.912 35.418 166.675 35.1102C166.438 34.8024 166.072 34.622 165.684 34.622V35.872ZM155.804 73V74.25C156.371 74.25 156.866 73.8689 157.012 73.3214L155.804 73ZM151.072 73L149.874 73.3559C150.031 73.8863 150.519 74.25 151.072 74.25V73ZM143.844 48.664L145.042 48.3081C145.04 48.2992 145.037 48.2903 145.034 48.2815L143.844 48.664ZM142.96 45.7L1 ...
```
Same here, the logos itself doesn't need to be touched. They were provided recently together with the color palette https://github.com/TooAngel/worlddriven/pull/158

TooAngel commented on static/index.html:
```diff
@@ -52,7 +52,7 @@
     </div>
     <div class="heading">
-      <img src="/static/images/globe.svg" class="logo" alt="World driven logo" height="60" width="60">
+      <img src="/static/images/favicon-32x32.png" class="logo" alt="World driven logo" height="60" width="60">
```
Well, don't use a low quality favicon as logo on the page, when you have an svg.

TooAngel commented on static/style.css:
```diff
@@ -124,11 +124,11 @@
 
 input:checked + .slider {
-  background-color: #2196F3;
+  background-color: #205572;
```
This is the direction the issue was meant. Maybe in the `index.html` the font color, ... could be adapted to the colors instead of plain black.

TooAngel reviewed (changes requested):
My intention of the issue is to bring more color on the page and update the page more according to the new logo and not changing the logo, sorry if I described it badly.

So replacing the current logo with the new one (use svg. not favicon as logo) and try to style the first page better fitting to the colors.

zbarovsky commented:
Omg I just realized I forgot to put the right svg for the title. Sorry about that I'll change it real quick

zbarovsky commented:
Ok, I added the logo_world_driven.svg as the logo and bumped the size up, so it should be good to go!

TooAngel commented:
Yes, perfect, that is the direction I thought of, thx.

Can you please squash your commits and make sure it is mergeable, there is a conflict in the index.html right now, and it is good to merge in my opinion.

TooAngel commented:
Ah, and can you add Fixes #174 to your Pull Request message, so that the issue is automatically closed as soon as you changes are merged, that would be cool.

zbarovsky commented:
Ok, I fixed the merge conflicts and added the 174 fix at the top. 

However, I couldn't get git squash to work effectively. I can keep trying if you want, but it kept adding commits, so it felt kind of redundant to keep adding commits to squash commits, sorry.

TooAngel commented:
I think you need to figure out how to squash the commits.

In this case, I would suggest rebasing.
You can read about the concept here: https://git-scm.com/book/en/v2/Git-Branching-Rebasing

What I would do:
Having your repository as remote `origin`
Add the tooangel/worlddriven as remote `worlddriven`: `git remote add worlddriven git@github.com:TooAngel/worlddriven.git`
Update the index: `git fetch --all --prune`
Start the rebase: `git rebase -i worlddriven/master` - This will put your commits on top of the commits of the worlddriven master branch and provide you with the option to squash, pick or fixup commits.
I guess you will see it in some kind of editor.
From the second line on replace the `pick` with `squash` - This will squash all commits into the first one.
After saving you will see the all commit messages, clean this up to a good message and save.
Last step force push the repository to your fork and brach `git push origin/development --force` (you rewrote history, so you have to use the force) (https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)

zbarovsky commented:
I did all the steps and the commits aren't squashing I'm really sorry. It also won't let me fetch your remote because I don't have permission, so I don't know if that has anything to do with it, but it's just not working on my end.

zbarovsky commented:
ok! I got it rebased and squashed and fixed the merge conflicts that we're still there!

TooAngel reviewed (approved):
Again thanks for contributing.

Close to perfect, you still have 2 commits, where the first one looks very messy and seems like a lot of changes that are reverted in the 2nd commit.
Thx for taking the time to look into the git rebasing thingy, I hope you helped you, too.
The actual diff of the PR looks fine.
Small things, the docker heading in the `README.md` doesn't need to be removed, some trailing whitespaces in the `index.html`.

Fine for me right now.

TooAngel commented:
@IanFindlay Would be cool, if you can review this one, too. If you are satisfied you can speed you the time to merge.

IanFindlay commented on static/index.html:
```diff
@@ -52,17 +52,15 @@
-    <h3>The Challenge</h3>
+    <h2 class='phases'>Phase One: Dynamic Code Ownership</h2>
 
+    <h3 class="subheading">The Challenge</h3>
```
This and the above 'phases' class is a nice idea, giving them different colours that are part of the colour scheme, but you've only done it on the first phase and the first subheading - This should be made consistent throughout the page.

IanFindlay commented on static/index.html:
```diff
@@ -52,17 +52,15 @@
     <div class="heading">
-      <img src="/static/images/globe.svg" class="logo" alt="World driven logo" height="60" width="60">
-      <h1>World Driven</h1>
+      <img src="/static/images/logo_world_driven.svg" class="logo" alt="World driven logo">
```
It would be nice if the white background of this svg was removed. Not strictly part of the issue this PR resolves, but it might be worth doing while were updating it anyway.

IanFindlay reviewed (changes requested):
Direction of the changes is right you just need to add the new classes throughout the page. Removing the background of the svg is a bonus that can be done later on a separate PR, so don't feel like you have to make that change for me to approve this PR.

TooAngel commented:
Regarding the commit history, I tried rebasing you branch, too, and I failed (too many conflicts and me being to lazy).

So I give you another approach:
Your repository `zbarovsky/development` is the `origin` remote. The `tooangel/worlddriven` is the `worlddriven` remote.
Checkout your branch. `git checkout development`
Update the index: `git fetch --all --prune`
Reset the commit history, but leave the workspace unchanged: `git reset worlddriven/master`
You can verify that your changes are still there: `git diff`
Remove the `README.md` change, I think they shouldn't be in there: `git checkout worlddriven/master README.md`
Commit your changes, as it was your first commit: `git commit -a` (Write or copy your commit message)
Push your changes: `git push origin/development -f` (Force, because we will rewrite history again)

I hope this helps.

zbarovsky commented:
Ok I'll get that all straightened out and pushed. Also that's real weird about the classes because I definitely applied that to all the headings I don't to know why it's only showing the first one.

IanFindlay commented:
> Also that's real weird about the classes because I definitely applied that to all the headings I don't to know why it's only showing the first one.

I suspected something strange might have happened there; don't worry about it and thank you again for your contribution.

zbarovsky commented:
Ok, I reset the git commits, and did a force push and got it to one clean commit. For some reason it keeps giving me merge conflicts over the class names in the index file. I even went line by line and lined everything up perfectly and it still errors over the class namings. So the correct class names should be on everything but it's the closest to one commit I think I can get.

IanFindlay reviewed (approved):
Thanks for making the changes. Looks good to me.

TooAngel commented:
@zbarovsky Well, with the last merge yours ran into a merge conflict, could please update yours one more time. Should go through this time.

zbarovsky commented:
ok I hit merge, it says this pull request can be automatically merged by project collaborators.

TooAngel commented:
I'm not sure how good you understood the world driven idea.

Pull Requests are automatically merged after a certain time, the current configuration is 5 days for a PR + additional 5 days per commit. (This gives the people enough time to review the Pull Request, and usually more commits mean a higher complexity).
The merge can be boosted by reviewing and approving the change. This will reduce the time to merge by a factor based on your previous contributions.

Pointed to your question, if you have more commits if will take longer until the Pull Request is merged (and can lead more often to merge conflicts).

I guess this Pull Request will be merged, depending on the number of commits in 1-3 days, so all good.
I would suggest trying to figure out how to squash it to one commit. Not necessary because of this PR, but for you to feel more comfortable with git and here you have a friendly playground.

TooAngel commented:
This pull request was merged by [worlddriven](https://www.worlddriven.org).
